packages/core/src/components/EmptyState/EmptyState.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { FC } from "react";
+import cx from "classnames";
+import Button from "../../components/Button/Button";
+import Link from "../../components/Link/Link";
+import Flex from "../../components/Flex/Flex";
+import Heading from "../../components/Heading/Heading";
+import Text from "../../components/Text/Text";
+import { EmptyStateProps, EmptyStateLayout, EmptyStateSupportingActionProps } from "./EmptyState.types";
+import styles from "./EmptyState.module.scss";
+
+const EmptyState: FC<EmptyStateProps> = ({
+  className,
+  title,
+  description,
+  illustration,
+  mainAction,
+  supportingAction,
+  layout = EmptyStateLayout.DEFAULT,
+  id,
+  "data-testid": dataTestId
+}) => {
+  const isCompact = layout === EmptyStateLayout.COMPACT;
+
+  return (
+    <Flex
+      direction={Flex.directions.COLUMN}
+      align={Flex.align.CENTER}
+      justify={Flex.justify.CENTER}
+      gap={isCompact ? Flex.gaps.SMALL : Flex.gaps.MEDIUM}
+      className={cx(styles.emptyState, styles[layout], className)}
+      id={id}
+      data-testid={dataTestId || "empty-state"}
```

Comment:
Let's use the constant instead of `"empty-state"` plus use the id like we do in other components